src/item_use.c

```diff
@@ -981,6 +982,13 @@ static const u8 sText_CantThrowPokeBall_TwoMons[] = _("Cannot throw a ball!\nThe
 static const u8 sText_CantThrowPokeBall_SemiInvulnerable[] = _("Cannot throw a ball!\nThere's no Pokémon in sight!\p");
 void ItemUseInBattle_PokeBall(u8 taskId)
 {
+    if (FlagGet(B_FLAG_NO_CATCHING))
+    {
+        static const u8 sText_BallsCannotBeUsed[] = _("Poké Balls cannot be used\nright now!\p");
+        DisplayItemMessage(taskId, 1, sText_BallsCannotBeUsed, CloseItemMessage);
+        return;
+    }
+    
```

Comment:
Considering this will only be set by the dev in debugging or at special points in game, this checks precedes all others and therefore makes its unnecessary to check the rest of the conditions which would happen if it would be handled like 'BALL_THROW_UNABLE_SEMI_INVULNERABLE' - so I would suggest leaving it as is

Reply:
Remember, this was changed so the flag could be used in other contexts than debugging.